src/viam/sdk/components/arm.hpp

```diff
@@ -0,0 +1,142 @@
+/// @file components/arm.hpp
+///
+/// @brief Defines an `Arm` component
+#pragma once
+
+#include <string>
+
+#include <boost/variant/variant.hpp>
+
+#include <viam/api/common/v1/common.pb.h>
+#include <viam/api/component/arm/v1/arm.pb.h>
+
+#include <viam/sdk/common/pose.hpp>
+#include <viam/sdk/resource/stoppable.hpp>
+#include <viam/sdk/spatialmath/geometry.hpp>
+
+namespace viam {
+namespace sdk {
+
+/// @defgroup Arm Classes related to the Arm component.
+
+/// @class Arm arm.hpp "components/arm.hpp"
+/// @brief An `Arm` represents a physical robot arm that exists in three-dimensional space.
+/// @ingroup Arm
+///
+/// This acts as an abstract parent class to be inherited from by any drivers representing
+/// specific arm implementations. This class cannot be used on its own.
+class Arm : public Component, public Stoppable {
+    // Base class for use below in defining kinematics data strong typedefs
+    template <class Tag>
+    struct RawBytes {
+        // Pre c++17 our derived classes aren't aggregate initializable so we need to define
+        // and using declare some ctors
+
+        RawBytes() = default;
+        RawBytes(std::vector<unsigned char> b) : bytes(std::move(b)) {}
+
+        std::vector<unsigned char> bytes{};
+    };
+
+    // Comparison operator helper for the data types below
+    template <class DataType>
+    struct EqCompare {
+        inline friend bool operator==(const DataType& lhs, const DataType& rhs) {
+            return lhs.bytes == rhs.bytes;
+        }
+    };
+
+   public:
+    struct KinematicsDataUnspecified : RawBytes<KinematicsDataUnspecified>,
+                                       EqCompare<KinematicsDataUnspecified> {};
+    struct KinematicsDataSVA : RawBytes<KinematicsDataSVA>, EqCompare<KinematicsDataSVA> {
+        using RawBytes<KinematicsDataSVA>::RawBytes;
+    };
+    struct KinematicsDataURDF : RawBytes<KinematicsDataURDF>, EqCompare<KinematicsDataURDF> {
+        using RawBytes<KinematicsDataURDF>::RawBytes;
+    };
+
+    /// @brief The kinematics of the component.
+    /// @returns The data in Viam's Spatial Vector Algebra (SVA) format, or URDF.
+    using KinematicsData =
+        boost::variant<KinematicsDataUnspecified, KinematicsDataSVA, KinematicsDataURDF>;
+
+    static KinematicsData from_proto(const viam::common::v1::GetKinematicsResponse& proto);
+
+    /// @brief Get the current position of the end of the arm.
+    inline pose get_end_position() {
+        return get_end_position({});
+    }
+
+    /// @brief Get the current position of the end of the arm.
+    /// @param pose The destination pose for the arm.
```

Comment:
(minor) this should be `@return` and below the `extra` param. also probably should be included with the documentation for the `inline` version as well for consistency's sake.